core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
I don't think just saying "safely" is informative or helpful. It would be helpful for future contributors if you give a short explanation about what exactly is expected from a comparator (which is nan == nan behavior for floats only) and why this *workaround* (which violates IEEE-754 standard, in a *particular way* to fit the need of the particular way dictionary hash behaves) is needed.